python/ray/tune/schedulers/resource_changing_scheduler.py

```diff
@@ -14,13 +17,337 @@
 logger = logging.getLogger(__name__)
 
 
-class _DistributeResources:
-    """Generic functionality for resource allocation functions"""
+@PublicAPI(stability="beta")
+class DistributeResources:
+    """This class creates a basic uniform resource allocation function.
 
-    def __init__(self, add_bundles: bool = False):
-        """If add_bundles is True, create new bundles from free resources.
-        Otherwise, spread them among base_trial_resource bundles."""
+    The function naively balances free resources (CPUs and GPUs) between
```

Comment:
Do you want to mention that users who wish to implement their own resource distribution policy could do it by extending this class. Maybe also point them to the few methods that are intended to facilitate modularization?

Reply:
Good idea, will do that